### PR TITLE
CT-3987 upgrade rails 5 to rails 6

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to upgrade rails 5 to rails 6 
## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [X] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets

https://dsdmoj.atlassian.net/jira/software/c/projects/CT/boards/25?modal=detail&selectedIssue=CT-3987

### CODE REVIEW

File: config/initializers/backtrace_silencers.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
 
 # You can add backtrace silencers for libraries that you're using but don't wish to see in your backtraces.
-# Rails.backtrace_cleaner.add_silencer { |line| line =~ /my_noisy_library/ }
+# Rails.backtrace_cleaner.add_silencer { |line| /my_noisy_library/.match?(line) }
 
-# You can also remove all the silencers if you're trying to debug a problem that might stem from framework code.
-# Rails.backtrace_cleaner.remove_silencers!
+# You can also remove all the silencers if you're trying to debug a problem that might stem from framework code
+# by setting BACKTRACE=1 before calling your invocation, like "BACKTRACE=1 ./bin/rails runner 'MyClass.perform'".
+Rails.backtrace_cleaner.remove_silencers! if ENV["BACKTRACE"]
```
Comment (willmcb) [inline config/initializers/backtrace_silencers.rb:8]: Nice to know this :)

File: config/initializers/filter_parameter_logging.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += [
+  :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
```
Comment (willmcb) [inline config/initializers/filter_parameter_logging.rb:5]: is this a typo here should it not be `:password` rather than `:passw`? :)

**Review by willmcb — CHANGES_REQUESTED** [review]
Please just check the comment about `:passw` other than that LGTM :)

File: config/initializers/filter_parameter_logging.rb (hunk 2 above)
Comment (ymao2) [reply config/initializers/filter_parameter_logging.rb:5]: thank you for double-checking this.  actually tbh not sure about it, this file was generated automatically when I run` rails app:update`, depending how the `filter_parameters`, as the` :password` is included in `:passw` in a way, let me find it out more

Comment (ymao2) [reply config/initializers/filter_parameter_logging.rb]: @willmcb,  couldn't find any place whether :passw includes key word of  'password' or not although I believe it should.  So I decided to add :password into the above list as well which make sure it did filter password key word out of log

Comment (ymao2) [reply config/initializers/filter_parameter_logging.rb]: @willmcb , I tested this PR on QA, it looks fine, could you approve it?  but I will find proper time to merge it

**Review by willmcb — APPROVED** [review]
LGTM 👍

**Pull request merged**